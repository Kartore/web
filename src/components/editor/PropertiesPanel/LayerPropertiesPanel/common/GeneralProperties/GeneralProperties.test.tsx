import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react';
import { describe } from 'vitest';

import * as Stories from './GeneralProperties.stories.tsx';

const { Default } = composeStories(Stories);

describe('Component: GeneralProperties', () => {
	describe('Snapshot', () => {
		it('Default', () => {
			const { asFragment } = render(<Default />);
			expect(asFragment()).toMatchSnapshot();
		});
	});
});
