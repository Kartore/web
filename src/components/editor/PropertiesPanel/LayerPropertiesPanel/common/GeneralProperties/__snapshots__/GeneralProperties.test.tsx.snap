// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component: GeneralProperties > Snapshot > Default 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-2 px-4"
  >
    <h3
      class="font-montserrat text-sm font-semibold"
    >
      General
    </h3>
    <div
      class="flex items-center justify-between text-sm"
    >
      <div
        class="font-semibold text-gray-600"
        id="react-aria-:r3:"
      >
        Source
      </div>
      <div
        aria-hidden="true"
        data-a11y-ignore="aria-hidden-focus"
        data-react-aria-prevent-focus="true"
        data-testid="hidden-select-container"
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          style="font-size: 16px;"
          tabindex="-1"
          type="text"
        />
        <label>
          Source
          <select
            tabindex="-1"
          >
            <option />
            <option
              value="openmaptiles"
            >
              openmaptiles
            </option>
            <option
              value="natural_earth_shaded_relief"
            >
              natural_earth_shaded_relief
            </option>
          </select>
        </label>
      </div>
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="react-aria-:r7: react-aria-:r3:"
        class="cursor-pointer active:bg-gray-300 flex w-1/2 flex-row items-center justify-between rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0 aria-expanded:bg-gray-200"
        id="react-aria-:r2:"
        type="button"
      >
        <p
          class="flex-1 overflow-hidden text-ellipsis text-start"
          id="react-aria-:r7:"
        >
          openmaptiles
        </p>
        <svg
          aria-hidden="true"
          class="w-4"
          viewBox="0 -960 960 960"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M480-360 280-560h400L480-360Z"
          />
        </svg>
      </button>
    </div>
    <div
      class="flex items-center justify-between text-sm"
    >
      <div
        class="font-semibold text-gray-600"
        id="react-aria-:rb:"
      >
        Source Layer
      </div>
      <div
        aria-hidden="true"
        data-a11y-ignore="aria-hidden-focus"
        data-react-aria-prevent-focus="true"
        data-testid="hidden-select-container"
        style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
      >
        <input
          style="font-size: 16px;"
          tabindex="-1"
          type="text"
        />
        <label>
          Source Layer
          <select
            tabindex="-1"
          >
            <option />
          </select>
        </label>
      </div>
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="react-aria-:rf: react-aria-:rb:"
        class="cursor-pointer active:bg-gray-300 flex w-1/2 flex-row items-center justify-between rounded border-none bg-gray-100 py-1 px-2 text-sm font-semibold transition-colors hover:bg-gray-200 focus-visible:bg-gray-200 focus-visible:outline-0 aria-expanded:bg-gray-200"
        id="react-aria-:ra:"
        type="button"
      >
        <p
          class="flex-1 overflow-hidden text-ellipsis text-start"
          id="react-aria-:rf:"
        />
        <svg
          aria-hidden="true"
          class="w-4"
          viewBox="0 -960 960 960"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M480-360 280-560h400L480-360Z"
          />
        </svg>
      </button>
    </div>
    <div
      aria-labelledby="react-aria-:rh:"
      class="h-auto w-full horizontal"
      id="react-aria-:rg:"
      role="group"
    >
      <div
        class="flex items-center justify-between"
      >
        <label
          class="text-sm font-semibold text-gray-600"
          id="react-aria-:rh:"
        >
          Zoom Range
        </label>
        <output
          aria-live="off"
          class="text-sm font-semibold text-gray-800"
          for="react-aria-:rh:-0 react-aria-:rh:-1"
        >
          0 - 24
        </output>
      </div>
      <div
        class="relative mt-2 h-0.5 w-full rounded-full bg-gray-300"
        style="position: relative; touch-action: none;"
      >
        <div
          class="absolute h-0.5 bg-gray-500"
          style="left: 0%; right: 0%;"
        />
        <div
          class="thumb mt-[1px] size-3 rounded-full border border-gray-500 bg-white"
          style="position: absolute; left: 0%; transform: translate(-50%, -50%); touch-action: none;"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              aria-describedby=""
              aria-details=""
              aria-labelledby="react-aria-:rh:"
              aria-orientation="horizontal"
              aria-valuetext="0"
              id="react-aria-:rh:-0"
              max="24"
              min="0"
              step="1"
              tabindex="0"
              type="range"
              value="0"
            />
          </div>
        </div>
        <div
          class="thumb mt-[1px] size-3 rounded-full border border-gray-500 bg-white"
          style="position: absolute; left: 100%; transform: translate(-50%, -50%); touch-action: none;"
        >
          <div
            style="border: 0px; clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;"
          >
            <input
              aria-describedby=""
              aria-details=""
              aria-labelledby="react-aria-:rh:"
              aria-orientation="horizontal"
              aria-valuetext="24"
              id="react-aria-:rh:-1"
              max="24"
              min="0"
              step="1"
              tabindex="0"
              type="range"
              value="24"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
